__init__.py

```diff
@@ -12,314 +12,420 @@
+        out = ""
+        if days > 0:
+            out += pronounce_number(days)
+            out += " days " if days == 1 else " day "
+        if hours > 0:
+            out += pronounce_number(hours)
+            out += " hour " if hours == 1 else " hours "
+        if minutes > 0:
+            out += pronounce_number(minutes)
+            out += " minute " if minutes == 1 else " minutes "
+        if seconds > 0:
+            out += pronounce_number(seconds)
+            out += " second" if seconds == 1 else " seconds"
+        return out
+    else:
+        # M:SS, MM:SS, H:MM:SS, Dd H:MM:SS format
+        out = ""
+        if days > 0:
+            out = str(days) + "d "
+        if hours > 0:
+            out += str(hours) + ":"
+        if minutes < 10 and hours > 0:
+            out += "0"
+        out += str(minutes)+":"
+        if seconds < 10:
+            out += "0"
+        out += str(seconds)
+        return out
+
+
+#####################################################################
 # TODO: add labels for timers, first, second, etc
-# TODO: add stop timer sound functionality
+
 class TimerSkill(MycroftSkill):
     def __init__(self):
         super(TimerSkill, self).__init__("TimerSkill")
         self.active_timers = []
-        self.should_converse = False
-        self.intent_context = None
-        self.stop_notify = False
-        self.allow_notify = False
         self.sound_file = join(abspath(dirname(__file__)), 'timerBeep.wav')
+        try:
+            self.eyes_fill = self.enclosure.eyes_fill
+        except:
+            self.eyes_fill = enclosure_eyes_fill
+        self.displaying_timer = None
+        self.beep_process = None
+        self.display_text = None
+        self.mute = False
 
     def initialize(self):
-        self.register_intent_file(
-            'start.timer.intent', self.handle_start_timer)
-        self.register_intent_file(
-            'status.timer.intent', self.handle_status_timer)
-        self.register_intent_file(
-            'cancel.timer.intent', self.handle_cancel_timer)
-        self.register_intent_file('stop.intent', self._stop)
         self.register_entity_file('duration.entity')
         self.register_entity_file('timervalue.entity')
         self.register_entity_file('all.entity')
 
-    def handle_start_timer(self, message):
-        """ callback for start timer intent
+        # Invoke update_display in one second to allow it to disable the
+        # cancel intent, since there are no timers to cancel yet!
+        self.schedule_repeating_event(self.update_display,
+                                      None, 1, name='ShowTimer')
 
-            Args:
-                message (Message): object passed by messagebus
-        """
+    # Handles 'Start a 30 second timer'
+    # TODO: Doesn't handle 'start 1 and a half minute timer'
+    @intent_file_handler('start.timer.intent')
+    def handle_start_timer(self, message):
         duration = message.data["duration"]
```

Comment:
errors out if the duration entity is not found. Having a get response to ask for the duration would be good here.